addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessResource.java

```diff
@@ -74,30 +74,65 @@ public NPMContentAccessResource( final ContentAccessHandler handler )
+                                 () -> uriInfo.getBaseUriBuilder()
+                                              .path( getClass() )
+                                              .path( packageName )
+                                              .build( NPM_PKG_KEY, type, name ) );
+    }
+
+    @ApiOperation( "Store NPM artifact content under the given artifact store (type/name), packageName and version." )
+    @ApiResponses( { @ApiResponse( code = 201, message = "Content was stored successfully" ), @ApiResponse( code = 400,
+                                                                                                            message = "No appropriate storage location was found in the specified store (this store, or a member if a group is specified)." ) } )
+    @PUT
+    @Path( "/{packageName}/{version}" )
+    public Response doCreate(
+            final @ApiParam( allowableValues = "hosted,group,remote", required = true ) @PathParam( "type" )
+                    String type, final @ApiParam( required = true ) @PathParam( "name" ) String name,
+            final @PathParam( "packageName" ) String packageName, final @PathParam( "version" ) String version,
+            final @Context UriInfo uriInfo, final @Context HttpServletRequest request )
+    {
+        final String path = Paths.get( packageName, PATH_SPLITER, version ).toString();
+        return handler.doCreate( NPM_PKG_KEY, type, name, path, request, new EventMetadata(),
+                                 () -> uriInfo.getBaseUriBuilder()
+                                              .path( getClass() )
+                                              .path( path )
+                                              .build( NPM_PKG_KEY, type, name ) );
+    }
+
+    @ApiOperation( "Store NPM artifact content under the given artifact store (type/name), packageName and tarball." )
+    @ApiResponses( { @ApiResponse( code = 201, message = "Content was stored successfully" ), @ApiResponse( code = 400,
+                                                                                                            message = "No appropriate storage location was found in the specified store (this store, or a member if a group is specified)." ) } )
+    @PUT
+    @Path( "/{packageName}/{cross}/{tarball}" )
+    public Response doCreate(
+            final @ApiParam( allowableValues = "hosted,group,remote", required = true ) @PathParam( "type" )
+                    String type, final @ApiParam( required = true ) @PathParam( "name" ) String name,
+            final @PathParam( "packageName" ) String packageName,
+            final @ApiParam( allowableValues = CROSS ) @PathParam( "cross" ) String cross,
+            final @PathParam( "tarball" ) String tarball, final @Context UriInfo uriInfo,
+            final @Context HttpServletRequest request )
+    {
+        final String path = Paths.get( packageName, PATH_SPLITER, CROSS, PATH_SPLITER, tarball ).toString();
```

Comment:
Again, same question about PATH_SPLITER as I asked above. Does this result in a `//` path part in there, or is the PATH_SPLITER parameter needed?